pkg/cmd/roachtest/roachtestutil/mixedversion/planner_test.go

```diff
@@ -303,11 +302,21 @@ func Test_upgradeTimeout(t *testing.T) {
 	assertTimeout(30*time.Minute, UpgradeTimeout(30*time.Minute)) // custom timeout applies.
 }
 
-func setBuildVersion() func() {
-	previousV := clusterupgrade.TestBuildVersion
+// setDefaultVersions overrides the test's view of the current build
+// as well as the oldest supported version. This allows the test
+// output to remain stable as new versions are released and/or we bump
+// the oldest supported version. Called by TestMain.
+func setDefaultVersions() func() {
```

Comment:
Nit: maybe `resetDefaultVersions` since that's how it's primarily used.

Reply:
Not sure I get it -- the `set` function sets test values and returns a "reset" function. If the function was called `reset`, it would seem like it's resetting back to original values?

Reply:
Fair. Although, `setDefaultVersions()()` is doing both of those things, i.e., setting and resetting :). I'm fine with the current name; I was mainly commenting to the fact that it's expected to be called from `TestMain`, i.e., not your typical setter.

Reply:
Ah I see. If the function was called with a `defer` (`defer setDefaultVersions()()`), I think the case for calling it `reset` would be stronger. But currently we do `reset := setDefaultVersions()` which IMO is clearer if we're assigning.